### PR TITLE
INTMDB-932: remove the default value of retain_backups_enabled

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description
Ticket: [CLOUDP-165348](https://jira.mongodb.org/browse/CLOUDP-165348), [INTMDB-932](https://jira.mongodb.org/browse/INTMDB-932)

This PR updates the deletion logic of the cluster after consulting the Atlas backup team:

- **Previous logic**: We set `retainBackup` to false as default if the user did not provide a value of the field
- **Correct logic**: We should set `retainBackup` only if the user provides a value for it. Otherwise, we should call the DELETE endpoint without passing any options.

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### TESTING
All the cluster acceptance tests are successful + I run a local test to check that the query param is sent only when `retain_backups_enabled` is provided

### CODE REVIEW

File: mongodbatlas/resource_mongodbatlas_cluster.go [hunk 1]
```diff
@@ -1081,10 +1078,6 @@ func resourceMongoDBAtlasClusterImportState(ctx context.Context, d *schema.Resou
 		return nil, fmt.Errorf("couldn't import cluster backup configuration %s in project %s, error: %s", *name, *projectID, err)
 	}
 
-	if err := d.Set("retain_backups_enabled", false); err != nil {
-		return nil, fmt.Errorf("couldn't import cluster backup configuration %s in project %s, error: %s", *name, *projectID, err)
-	}
-
```
Comment (andreaangiolillo) [inline mongodbatlas/resource_mongodbatlas_cluster.go:1087]: we should not set `retain_backups_enabled` as part of the import but only during cluster creation as the GET CLUSTER endpoint does not return `retain_backups_enabled`

File: mongodbatlas/resource_mongodbatlas_advanced_cluster.go [hunk 2]
```diff
@@ -842,10 +839,6 @@ func resourceMongoDBAtlasAdvancedClusterImportState(ctx context.Context, d *sche
 		log.Printf(errorClusterAdvancedSetting, "name", u.ID, err)
 	}
 
-	if err := d.Set("retain_backups_enabled", false); err != nil {
-		return nil, fmt.Errorf(errorClusterAdvancedSetting, "retain_backups_enabled", u.ID, err)
-	}
-
```
Comment (andreaangiolillo) [inline mongodbatlas/resource_mongodbatlas_advanced_cluster.go:848]: we should not set `retain_backups_enabled` as part of the import but only during cluster creation as the GET CLUSTER endpoint does not return `retain_backups_enabled`

Comment (gssbzn) [reply mongodbatlas/resource_mongodbatlas_advanced_cluster.go:848]: should we open a bug for this?

**Review by gssbzn — APPROVED** [review]

File: mongodbatlas/resource_mongodbatlas_advanced_cluster.go (hunk 2 above)
Comment (andreaangiolillo) [reply mongodbatlas/resource_mongodbatlas_advanced_cluster.go:848]: sure, created this ticket to keep track of this https://jira.mongodb.org/browse/INTMDB-932

**Review by martinstibbe — APPROVED** [review]
LGTM

**Review by Zuhairahmed — APPROVED** [review]

**Pull request merged**